transformation/transformation.go

```diff
@@ -0,0 +1,703 @@
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Bool {
+		return firstInput.value.(bool) == secondInput.value.(bool), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Bytes {
+		return string(firstInput.value.([]byte)) == string(secondInput.value.([]byte)), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Date {
+		return firstInput.value.(civil.Date) == secondInput.value.(civil.Date), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Timestamp {
+		return firstInput.value.(time.Time).Equal(secondInput.value.(time.Time)), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.String {
+		return firstInput.value == secondInput.value, nil
+	}
+	return false, fmt.Errorf("unsupported data types for comparison: %T, %T", firstInput.value, secondInput.value)
+}
+
+func compareGreaterThan(firstInput, secondInput inputValue) (bool, error) {
+	if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Int64 {
+		return firstInput.value.(int64) > secondInput.value.(int64), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Float64 {
+		return firstInput.value.(float64) > secondInput.value.(float64), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Date {
+		return firstInput.value.(civil.Date).After(secondInput.value.(civil.Date)), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Timestamp {
+		return firstInput.value.(time.Time).After(secondInput.value.(time.Time)), nil
+	}
+	return false, fmt.Errorf("unsupported data types for comparison: %T, %T", firstInput.value, secondInput.value)
+}
+
+func compareLessThan(firstInput, secondInput inputValue) (bool, error) {
+	if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Int64 {
+		return firstInput.value.(int64) < secondInput.value.(int64), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Float64 {
+		return firstInput.value.(float64) < secondInput.value.(float64), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Date {
+		return firstInput.value.(civil.Date).Before(secondInput.value.(civil.Date)), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Timestamp {
+		return firstInput.value.(time.Time).Before(secondInput.value.(time.Time)), nil
+	}
+	return false, fmt.Errorf("unsupported data types for comparison: %T, %T", firstInput.value, secondInput.value)
+}
+
+func applyMathOp(firstInput, operator, secondInput inputValue) (interface{}, error) {
+	switch operator.value.(string) {
+	case AddOperator:
+		switch firstInput.value.(type) {
+		case int64:
+			switch secondInput.value.(type) {
+			case int64:
```

Comment:
same as above - the output of this should be based on the expected output type.
So we can run int + float type operations.